Misc fixes for kubecf

mook-as opened the pull request:
- We do no have a separate `ssh-proxy` docker image; use the `ssh` image instead in the chart.
- Enable RBAC by default
- Adjust the concourse pipeline to also replace the docker repository (in addition to the tag) when packaging the helm chart, so that we do not need to override anything by default.
- Adjust the helm chart version to include enough identifiers to always be sortable:
  - The number of commits (only counting merges &c.), so that newer charts (of the same tag) always sort later
  - The date, so that rebuilds (e.g. changed docker image version) sort later.
  - Also include, as non-sorting-involved metadata, a hash of the image tags/versions.

mook-as committed:
fix: ssh-proxy uses the ssh docker image

mook-as committed:
feat: concourse: override docker repo in chart

In the concourse pipeline, also override the concourse repository
(instead of just the tag) in the emitted values.yaml.  This ensures that
the user will need to do nothing by default to use the generated chart
(since we upload the docker images to the given docker repository).

mook-as committed:
fix: chart: enable RBAC by default

This matches what kubecf needs; as the only consumer (for now) that
makes a sane default.

mook-as committed:
fix: package: normalize version

This makes sure the version string is always semver-comliant and always
has a commits-since-empty count that is increasing.  This way helm can
actually sort the builds correctly and pick up the newest build.

mook-as committed:
feat: package: add hash of override to version

When packaging, add the hash of the (rendered) overrides YAML file to
the version string, so that we get different versions when we update the
docker images involved.  Also include the time stamp so that helm can
sort things appropriately.

viovanov commented on chart/templates/ssh-proxy.yaml:
```diff
@@ -49,7 +49,8 @@ spec:
       - name: {{ $component }}
         {{- with index .Values $component "image" }}
-        image: {{ printf "%s/eirinix-%s:%s" .repository $component .tag | quote }}
+        # Note: this is in the "ssh" image, not "ssh-proxy".
```
I don't understand the comment.
What is "in the image"?

viovanov commented on .concourse/pipeline.yaml.gomplate:
```diff
@@ -222,9 +225,27 @@ jobs:
+        {{- if . | strings.HasSuffix "-setup" | not }}
+        {{ . }}:
+          image:
+            repository: {{ $.docker.org }}
```
Should be `.docker.repository`

viovanov commented on .concourse/pipeline.yaml.gomplate:
```diff
@@ -222,9 +225,27 @@ jobs:
+            tag: <%= File.read('../s3.{{ . }}-version/version') %>
+          {{- if printf "%s-setup" . | has $images }}
+          setup-image:
+            repository: {{ $.docker.org }}
```
Should be `.docker.repository`

mook-as committed:
fix: address review comments in #5

- https://github.com/SUSE/eirinix-helm-release/pull/5#discussion_r419788919
  Expand the comment to be clear what the issue is
- https://github.com/SUSE/eirinix-helm-release/pull/5#discussion_r419789211
- https://github.com/SUSE/eirinix-helm-release/pull/5#discussion_r419789285
  Rename concourse variable `.docker.org` to `.docker.repository` for
  clarity.
